src/realm/query_conditions.hpp

```diff
@@ -73,8 +73,8 @@ struct Contains : public HackClass {
         if (m1.is_null())
             return !m2.is_null();
         if (m1.is_type(type_String, type_Binary) && Mixed::types_are_comparable(m1, m2)) {
-            BinaryData b1 = m1.get_binary();
-            BinaryData b2 = m2.get_binary();
+            BinaryData b1 = m1.export_to_type<BinaryData>();
```

Comment:
I've updated these query conditions to do more explicit type checking instead of relying on `Mixed::types_are_compatible()`. That should speed things up a bit.